File: src/main/java/org/example/GlobalData.java
package org.example;

public class GlobalData {
    private double simulationTime;
    private double simulationStepTime;
    private double conductivity;
    private double alpha;
    private double tot;
    private double initialTemp;
    private double density;
    private double specificHeat;
    private int nN;
    private int nE;
    private Grid grid;

    public GlobalData(double simulationTime, double simulationStepTime, double conductivity, double alpha,
                      double tot, double initialTemp, double density, double specificHeat, int nN, int nE) {
        this.simulationTime = simulationTime;
        this.simulationStepTime = simulationStepTime;
        this.conductivity = conductivity;
        this.alpha = alpha;
        this.tot = tot;
        this.initialTemp = initialTemp;
        this.density = density;
        this.specificHeat = specificHeat;
        this.nN = nN;
        this.nE = nE;
        this.grid = new Grid(nN, nE);
    }

    public double getSimulationTime() {
        return simulationTime;
    }

    public void setSimulationTime(double simulationTime) {
        this.simulationTime = simulationTime;
    }

    public double getSimulationStepTime() {
        return simulationStepTime;
    }

    public void setSimulationStepTime(double simulationStepTime) {
        this.simulationStepTime = simulationStepTime;
    }

    public double getConductivity() {
        return conductivity;
    }

    public void setConductivity(double conductivity) {
        this.conductivity = conductivity;
    }

    public double getAlpha() {
        return alpha;
    }

    public void setAlpha(double alpha) {
        this.alpha = alpha;
    }

    public double getTot() {
        return tot;
    }

    public void setTot(double tot) {
        this.tot = tot;
    }

    public double getInitialTemp() {
        return initialTemp;
    }

    public void setInitialTemp(double initialTemp) {
        this.initialTemp = initialTemp;
    }

    public double getDensity() {
        return density;
    }

    public void setDensity(double density) {
        this.density = density;
    }

    public double getSpecificHeat() {
        return specificHeat;
    }

    public void setSpecificHeat(double specificHeat) {
        this.specificHeat = specificHeat;
    }

    public int getnN() {
        return nN;
    }

    public void setnN(int nN) {
        this.nN = nN;
    }

    public int getnE() {
        return nE;
    }

    public void setnE(int nE) {
        this.nE = nE;
    }

    public Grid getGrid() {
        return grid;
    }

    public void setGrid(Grid grid) {
        this.grid = grid;
    }

    @Override
    public String toString() {
        return "GlobalData{" +
                "simulationTime=" + simulationTime +
                ", simulationStepTime=" + simulationStepTime +
                ", conductivity=" + conductivity +
                ", alpha=" + alpha +
                ", tot=" + tot +
                ", initialTemp=" + initialTemp +
                ", density=" + density +
                ", specificHeat=" + specificHeat +
                ", nN=" + nN +
                ", nE=" + nE +
                '}';
    }
}

File: GlobalData.py
class GlobalData:
    def __init__(self, variables):
        self.simulationTime = variables[find_value(variables, 'SimulationTime')][1]
        self.simulationStepTime = variables[find_value(variables, 'SimulationStepTime')][1]
        self.conductivity = variables[find_value(variables, 'Conductivity')][1]
        self.alpha = variables[find_value(variables, 'Alfa')][1]
        self.ambientTemp = variables[find_value(variables, 'Tot')][1]
        self.initialTemp = variables[find_value(variables, 'InitialTemp')][1]
        self.density = variables[find_value(variables, 'Density')][1]
        self.specificHeat = variables[find_value(variables, 'SpecificHeat')][1]
        self.nN = variables[find_value(variables, 'Nodes number')][1]
        self.nE = variables[find_value(variables, 'Elements number')][1]

    def __str__(self):
        return (f"""
        SimulationTime = {self.simulationTime}, 
        SimulationStepTime = {self.simulationStepTime}, 
        Conductivity = {self.conductivity}, 
        Alfa = {self.alpha}, 
        Tot = {self.ambientTemp}, 
        InitialTemp = {self.initialTemp}, 
        Density = {self.density}, 
        SpecificHeat = {self.specificHeat}, 
        Nodes number = {self.nN}, 
        Elements number = {self.nE}
        """)

def find_value(array, value):
    for i in range(len(array)):
        if array[i][0] == value:
            return i